docs/spec/concepts.rst

```diff
@@ -3,97 +3,174 @@
 Type system concepts
 ====================
 
-.. _`union-types`:
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it will check the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic will be.  Python was
+historically a dynamically-typed language.
+
+**Gradual typing** is the name for a specific way to combine static and dynamic
+typing.  It allows mixing static and dynamic checking at a fine level of
+granularity.  For instance, in Python, individual variables, parameters, and
+return values can be either statically or dynamically typed.  Data structures
+such as objects can have a mix of static and dynamic checking.  As an example,
+a Python dictionary could choose to have static checking of the key type but
+dynamic checking of the value type.
+
+In gradual typing, a dynamically typed value is indicated by a special
+"unknown" or "dynamic" type.  In Python, the unknown type is spelled
+:ref:`Any`. It indicates to the static type checker that this value should not
+be subject to static checking.  The system should not signal a static type
+error for use of an expression with type :ref:`Any`.  Instead, the expression's
+value will be dynamically checked, according to the Python runtime's usual
+checks on the types of runtime values.
+
+This specification describes a gradual type system for Python.
+
+Static, dynamic, and gradual types
+----------------------------------
+
+We will refer to types that do not contain :ref:`Any` as a sub-part as **static
```

Comment:
Yeah, I tried something different here to avoid the awkwardness of "fully static type," but I agree that it didn't work out that well, and isn't consistent with the literature. I like your suggestion, I'll try a re-wording to use the terms in that way.